python/singa/autograd.py

```diff
@@ -2522,3 +2523,144 @@ def backward(self, dy):
 def max(a,b):
     return Max()(a,b)[0]
 
+
+class And(Operation):
+    def __init__(self):
+        super(And, self).__init__()
+
+    def forward(self, a, b):
+        m = singa.__div__(a,b)
+
+        mask0 = singa.GEFloat(m,1)
+        mask1 = singa.LEFloat(m,1)
+        cur = singa.__mul__(mask0,mask1)
+
+        return cur
+
+    def backward(self, dy):
+        assert 0,('no gradient')
+        return None
+
+def _and(a,b):
+    return And()(a,b)[0]
+
+
+class Or(Operation):
+    def __init__(self):
+        super(Or, self).__init__()
+
+    def forward(self, a, b):
+        #find equal element-wise
+        m = singa.__sub__(a,b)
```

Comment:
not clear, plz use this:
```
        m = singa.__add__(singa.PowFloat(singa.Sign(a), 2.0), singa.PowFloat(singa.Sign(b), 2.0))
        cur = singa.Sign(m)
```

Reply:
i think the equation is not correct if a = -b (element-wise)

Reply:
if a = -b, the Signs of them are, 1, and -1, and then, you take a square on it, the result will be 1 and 1, and you add them together, is 2, take the Sign again, you get a 1 finally.

Reply:
ok got it. just changed the code already.